velox/functions/sparksql/UnscaledValueFunction.cpp

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "velox/functions/sparksql/UnscaledValueFunction.h"
+
+#include "velox/expression/DecodedArgs.h"
+
+namespace facebook::velox::functions::sparksql {
+namespace {
+
+// Return the unscaled bigint value of a decimal, assuming it
+// fits in a bigint. Only short decimal input is accepted.
+class UnscaledValueFunction final : public exec::VectorFunction {
+  void apply(
+      const SelectivityVector& rows,
+      std::vector<VectorPtr>& args,
+      const TypePtr& outputType,
+      exec::EvalCtx& context,
+      VectorPtr& result) const final {
+    VELOX_USER_CHECK(
+        args[0]->type()->isShortDecimal(),
+        fmt::format(
```

Comment:
drop fmt::format

This check is redundant. Consider removing it.

Reply:
Currently, Velox can only restrict input to decimal values, but it cannot restrict it to short decimal. Therefore, this validation has been added.